src/server/service/kv.rs

```diff
@@ -926,10 +922,17 @@ impl<T: RaftStoreRouter<RocksEngine> + 'static, E: Engine, L: LockManager> Tikv
                     req,
                     &tx,
                 );
+                req_len -= 1;
+                if enable_req_batch
+                    && batcher.is_none()
+                    && req_len >= MAX_BATCH_GET_REQUEST_COUNT
+                    && storage.is_readpool_busy()
+                {
+                    batcher = Some(ReqBatcher::new());
+                }
             }
             if let Some(mut batch) = batcher {
```

Comment:
Can the batcher be turned off once the read pressure is released?

Reply:
Yes... You can see that I just created one batcher for every BatchCommandsRequest